### PR TITLE
PKborgs can no longer activate projectile fields if someone is buckled

### PR DESCRIPTION
**Pull request by silicons**
complement to #32251

### CODE REVIEW

**Comment by Arianya** [comment]
Isn't this a balance/rebalance or a tweak, since it wasn't intended behaviour in the first place?

Is oranges using GBP to reward kevinz for doing his dirty work??

brb calling journalist

**Comment by Robustin** [comment]
I'm suing this PR for GBP Fraud.

**Comment by ExcessiveUseOfCobblestone** [comment]
how many gbp does this give him?

**Comment by Mercenaryblue** [comment]
Its only missing the Honk tag.

**Comment by optimumtact** [comment]
-81

**Comment by Arianya** [comment]
I would have presumed that PRB: No Update would take priority over PRB: Reset but that might be faulty

**Comment by Armhulen** [comment]
finally i won't be at the bottom ok quickly merge it

**Comment by davethwave** [comment]
how did this happen?

**Comment by calzilla1** [comment]
Citrus machine [B]roke

**Comment by SamSamsonthe3rd** [comment]
You see, this is why this is the premier meme server.

**Comment by kevinz000** [comment]
NO

File: code/game/objects/items/robot/robot_items.dm [hunk 1]
```diff
@@ -546,11 +546,15 @@
 		to_chat(user, "<span class='boldwarning'>[src] is still recycling its projectors!</span>")
 		return
 	cycle_delay = world.time + PKBORG_DAMPEN_CYCLE_DELAY
-	active = !active
-	if(active)
-		activate_field(user)
+	if(!active)
+		if(!user.has_buckled_mobs())
+			activate_field()
+			active = TRUE
```
Comment (ninjanomnom) [inline code/game/objects/items/robot/robot_items.dm]: Move this into activate/deactivate field respectively

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/game/objects/items/robot/robot_items.dm [hunk 2]
```diff
@@ -546,11 +546,15 @@
...
-		activate_field(user)
+	if(!active)
+		if(!user.has_buckled_mobs())
+			activate_field()
+			active = TRUE
+		else
+			to_chat(user, "<span class='warning'>[src] refuses to activate!</span>")
 	else
 		deactivate_field()
+		active = FALSE
```
Comment (ninjanomnom) [inline code/game/objects/items/robot/robot_items.dm]: See above

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

**Review by ninjanomnom — DISMISSED** [review]

**Comment by lzimann** [comment]
conflicts @kevinz000

File: code/game/objects/items/robot/robot_items.dm [hunk 3]
```diff
@@ -546,9 +546,11 @@
...
 		return
 	cycle_delay = world.time + PKBORG_DAMPEN_CYCLE_DELAY
-	active = !active
-	if(active)
-		activate_field(user)
+	if(!active)
+		if(!user.has_buckled_mobs())
+			activate_field()
+		else
+			to_chat(user, "<span class='warning'>[src] refuses to activate!</span>")
```
Comment (lzimann) [inline code/game/objects/items/robot/robot_items.dm]: I feel like this should give a reason on why it refuses to activate.

Comment (kevinz000) [reply code/game/objects/items/robot/robot_items.dm]: kk

**Review by ninjanomnom — APPROVED** [review]

**Pull request merged**